### DISCUSSION TITLE
Moving Craft to a new PHP framework

### DISCUSSION THREAD

**Discussion by ma3ain** [craftcms/cms#13219] [General]
I have seen [Brandon's answer on Stack Exchange](https://craftcms.stackexchange.com/a/40794) and I would like to share my idea:

The PHP community and language seem to be struggling. Recently, the PHP foundation was established to support and develop this regard.

I know how hard it is to make the transition to a new framework. Although Laravel/Symfony have been around for a while, I recently saw [a new framework worth considering](https://trongate.io/). It has [a unique structure (Modular HAVC)](https://trongate.io/docs/information/introducing-modular-havc) that makes it fast and light. Trongate's top priority is stability. Although it is somewhat new and still needs a lot of contributions and contributors, it is promising.

Just wanted to point out how great Trongate is and if it might be considered.

**Comment by brandonkelly** [comment]
Think we will stick with something more established. Thanks for the suggestion though!

**Reply by vitalijalbu** [reply]
Hi Brandon, I’m also interested in this migration, I know there will be some concurrency with **Statamic** but I remember you said that one day you will move away… any news about this idea? Maybe on v6??🤝💪

**Reply by brandonkelly** [reply]
Yep, we’re switching to Laravel in v6.

**Reply by ghost** [reply]
> Yep, we’re switching to Laravel in v6.

do you ahve any idea of tools/lnguages you'll use? like **inertia**, **vue**, **blade**, **livewire**?
Thanks

**Reply by Anubarak** [reply]
@brandonkelly are you planning to work around the snake case assumption in laravel? Or at least provide a way to overwrite enough things to have fields/attributes/queries as we like so we don't have to build our code style / naming conventions around the underlying framework?

The requirement to have the database/queries and attributes in snake case in laravel is what always prevented me to use laravel in favour of Symfony.
I really love craft and it's current naming conventions even the db column naming. I really don't want see snake case hell in my code and I would highly appreciate if you could include a way to use things like

https://stackoverflow.com/a/70673855/5604055
